deep_regression.py

```diff
@@ -0,0 +1,251 @@
+            with slim.arg_scope([slim.fully_connected],
+                                activation_fn=tf.nn.relu,
+                                weights_regularizer=slim.l2_regularizer(BETA)):
+                net = slim.stack(inputs,
+                                 slim.fully_connected,
+                                 HIDDEN_LAYERS,
+                                 scope='fc')
+                end_points['fc'] = net
+                predictions = slim.fully_connected(net, 1, activation_fn=None,
+                                                   scope='prediction')
+                end_points['out'] = predictions
+                return predictions, end_points
+
+    # tf.Graph input
+    x = tf.placeholder("float", [None, input_features], name='features')
+    y = tf.placeholder("float", [None], name='labels')
+
+    # Construct the grpah for forward pass
+    pred, end_points = multilayer_neural_network_model(x, HIDDEN_LAYERS, BETA)
+
+    # Print name and shape of each tensor.
+    print("Layers")
+    for k, v in end_points.items():
+        print('name = {}, shape = {}'.format(v.name, v.get_shape()))
+
+    # Print name and shape of parameter nodes  (values not yet initialized)
+    print("Parameters")
+    for v in slim.get_model_variables():
+        print('name = {}, shape = {}'.format(v.name, v.get_shape()))
+
+    # Create the global step for monitoring the learning_rate and training.
+    global_step = get_or_create_global_step()
+
+    # Define your exponentially decaying learning rate
+    lr = tf.train.exponential_decay(learning_rate=INITIAL_LEARNING_RATE,
+                                    global_step=global_step,
+                                    decay_steps=decay_steps,
+                                    decay_rate=LEARNING_RATE_DECAY_FACTOR,
+                                    staircase=True)
+
+# ********************** LOSS && OPTIMIZE *************************************
+
+    mean_squared_error_loss = tf.reduce_sum(
+        tf.square(tf.transpose(pred) - y),
+        name="loss") / (2 * total_observations)
+
+    regularization_loss = tf.add_n(tf.losses.get_regularization_losses())
+
+    total_loss = mean_squared_error_loss + regularization_loss
+    # total_loss = tf.losses.get_total_loss(add_regularization_losses=True)
+
+#                   ************************************
+
+    optimizer = tf.train.AdamOptimizer(learning_rate=lr).\
+        minimize(total_loss, global_step=global_step)
+
+# ******************************************************************************
+    # Launch the graph
+    with tf.Session() as sess:
+
+        sess.run(tf.global_variables_initializer())
+
+        # Create a saver object which will save all the variables
+        saver = tf.train.Saver()
+
+        for epoch in range(TRAINING_EPOCHS + 1):
+            avg_cost = 0.0
+            total_batch = int(total_observations / BATCH_SIZE)
+            for i in range(total_batch - 1):    # Loop over all batches
```

Comment:
use tf.train.shuffle_batch, so you don't have to do shuffling and batching manually, and reduce your code size.